web/libs/editor/src/regions/Result.js

```diff
@@ -201,12 +219,8 @@ const Result = types
         return true;
       };
 
-      if (control.visiblewhen === "choice-selected") {
-        return isChoiceSelected();
-      }
-      if (control.visiblewhen === "choice-unselected") {
-        return !isChoiceSelected();
-      }
+      if (findNodeWithVisibleWhen(control, "choice-selected")) return control.isVisible && isChoiceSelected();
+      if (findNodeWithVisibleWhen(control, "choice-unselected")) return control.isVisible && !isChoiceSelected();
```

Comment:
As I see, there are `perregion` checks in `isChoiceSelected`. It needs to be checked, but most probably we cannot use `control.isVisible` in that case. As it checks only the current visible state of the control and perregions can contain more that one result which isVisible will not take into account.

Reply:
Good catch @Gondragos, thanks. I am checking it now. If you have in mind any particular template where we can reproduce this specific case is more than welcome (cc: @hlomzik )